src/main/java/bio/terra/cli/command/datarefs/CheckAccess.java

```diff
@@ -1,52 +1,78 @@
 package bio.terra.cli.command.datarefs;
 
-import bio.terra.cli.auth.AuthenticationManager;
+import bio.terra.cli.command.helperclasses.CommandSetup;
+import bio.terra.cli.command.helperclasses.FormatFlag;
 import bio.terra.cli.context.CloudResource;
-import bio.terra.cli.context.GlobalContext;
 import bio.terra.cli.context.TerraUser;
-import bio.terra.cli.context.WorkspaceContext;
 import bio.terra.cli.service.WorkspaceManager;
-import java.util.concurrent.Callable;
 import picocli.CommandLine;
 
 /** This class corresponds to the third-level "terra data-refs check-access" command. */
 @CommandLine.Command(
     name = "check-access",
     description = "Check if you have access to a data reference.")
-public class CheckAccess implements Callable<Integer> {
+public class CheckAccess extends CommandSetup {
 
   @CommandLine.Option(
       names = "--name",
       required = true,
       description = "The name of the data reference, scoped to the workspace.")
   private String name;
 
-  @Override
-  public Integer call() {
-    GlobalContext globalContext = GlobalContext.readFromFile();
-    WorkspaceContext workspaceContext = WorkspaceContext.readFromFile();
+  @CommandLine.Mixin FormatFlag formatFlag;
 
-    new AuthenticationManager(globalContext, workspaceContext).loginTerraUser();
+  /** Check if the user and their proxy group have access to a data reference. */
+  @Override
+  protected void execute() {
     CloudResource dataReference =
         new WorkspaceManager(globalContext, workspaceContext).getDataReference(name);
 
     TerraUser currentTerraUser = globalContext.requireCurrentTerraUser();
     boolean userHasAccess = dataReference.checkAccessForUser(currentTerraUser, workspaceContext);
-    boolean petSaHasAccess = dataReference.checkAccessForPetSa(currentTerraUser, workspaceContext);
+    boolean proxyGroupHasAccess =
+        dataReference.checkAccessForPetSa(currentTerraUser, workspaceContext);
+
+    CheckAccessReturnValue checkAccessReturnValue =
+        new CheckAccessReturnValue(userHasAccess, proxyGroupHasAccess);
+    formatFlag.printReturnValue(
+        checkAccessReturnValue,
+        returnValue -> {
+          this.printText(returnValue);
```

Comment:
nit: `CheckAccess::printText`?

Reply:
That method isn't `static`, so I don't think I can use that nicer notation. The reason it's not `static` is that it uses the global context object directly. I could have changed it to just use the return value object, but I thought it was useful to include the emails in just the text output, because the same information is already available for scripting in the `auth status` json output.

Reply:
for future, for non-static methods I think `this::printText` should work.

Reply:
Didn't know about that, thanks! Will update in the coming (part 2) PR.